kev1n1in/AsiaYoTravelMap
# feature/homepage

## Description
This PR completes the basic functionality for the homepage in the first version. The following features and implementations have been added:

Data Fetching: The homepage fetches user-specific data from Firebase using the user's uid. This ensures that only the journeys related to the authenticated user are displayed.

Journey Time Display: Each journey card on the homepage now displays the timeline, showing the start and end times for each journey. The times are calculated based on the dates and times of the journey events.

Deletion and Refetch: When a user deletes a journey, the list automatically refetches the data to reflect the updated state, ensuring the UI is always in sync with the database.

UI Enhancements: Added animations to the delete button using Framer Motion for a better user experience.

## Test Plan

1. Journey Time Display:

- Check that each journey card shows the correct start and end times.
- Confirm that times are displayed accurately according to the data stored in Firebase. 

2. Journey Deletion and Refetch:

- Click on the delete button for any journey.
- Confirm that a confirmation dialog appears.
- After confirming deletion, ensure the journey is removed from the list and the list is refetched.

3.UI and Animation:

- Hover over and click the delete button to verify that animations are working as expected.
- Ensure that all elements are responsive and behave correctly .
